internal/sink/sink.go

```diff
@@ -97,33 +100,37 @@ func NewSink(c Config, logger *zap.Logger, listener ancla.Register) Sink {
 						deliveryInterval: c.DeliveryInterval,
 						deliveryRetries:  c.DeliveryRetries,
 					},
+					client: &http.Client{},
 				}
+				if len(wh.ReceiverURLs) == 0 && len(wh.DNSSrvRecord.FQDNs) == 0 {
+					return nil, fmt.Errorf("either `ReceiverURLs` or `DNSSrvRecord` must be used")
+				} else if len(wh.ReceiverURLs) > 0 && len(wh.DNSSrvRecord.FQDNs) > 0 {
+					return nil, fmt.Errorf("either `ReceiverURLs` or `DNSSrvRecord` must be used but not both")
+				}
+
 				if len(wh.ReceiverURLs) > 0 {
 					urlCount, err := getUrls(wh.ReceiverURLs[1:])
 					if err != nil {
-						v2.logger.Error("error recevied parsing urls", zap.Error(err))
+						return nil, errors.Join(err, fmt.Errorf("error recevied parsing urls"))
 					}
 					v2.updateUrls(urlCount, wh.ReceiverURLs[0], wh.ReceiverURLs[1:])
-				} else {
-					sortBy := wh.DNSSrvRecord.LoadBalancingScheme
-					for _, domain := range wh.DNSSrvRecord.FQDNs {
-						dialer := new(Dialer)
-						err := dialer.lookup(domain, sortBy)
-						if err != nil {
-							v2.logger.Error("error received looking up service records", zap.Error(err))
-						}
-						dialer.NewClient()
-						v2.dialers = append(v2.dialers, dialer)
-					}
 				}
+
+				transport, err := NewSRVRecordDailer(wh.DNSSrvRecord)
```

Comment:
I know it was already mentioned but just making a note so we remember to change this to NewSRVRecordDialer